### PR TITLE
Added an API to store information when a new device is connected

### LINKED ISSUE #727: Create an endpoint to store the device's (smart speaker) information in the User's accounting model
As discussed in the meeting, the list of connected devices are supposed to be displayed on the web client. Hence, there needs to be an API implemented which when called from the Android/iOS devices on SignUp using Wifi Direct, will store the information of the connected device (Smart speaker). The information will have to be stored in the accounting model of the particular User.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by Akshat-Jain**
Fixes #727 

Changes: I've added a servlet whose endpoint, that is _/aaa/addNewDevice.json_, returns the information of the connected device (Smart Speaker) in a JSON format and also store that information in a file for future use. This stored information can then be retrieved and used to display the list of connected devices on the web client (as discussed in the meeting).

### CODE REVIEW

**Comment by Akshat-Jain** [comment]
@dynamitechetan @DravitLochan @saurabhjn76 Please review. Thanks :-)

**Comment by Akshat-Jain** [comment]
@arundhati24 @batbrain7 @imjog Please review :-)

File: src/ai/susi/SusiServer.java [hunk 1]
```diff
@@ -472,6 +472,7 @@ private static void setServerHandler(File dataFile){
                 TopMenuService.class,
                 PasswordChangeService.class,
                 ListSettingsService.class,
+								AddNewDevice.class,
```
Comment (PrP-11) [inline src/ai/susi/SusiServer.java]: Please check indentation here. :-)

**Comment by Akshat-Jain** [comment]
@dynamitechetan I've made the requested changes. Please review :-)
Thanks :-)

File: src/ai/susi/server/api/aaa/AddNewDevice.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+public class AddNewDevice extends AbstractAPIHandler implements APIHandler{
+
+    @Override
+    public String getAPIPath() {
+        return "/aaa/addNewDevice.json";
+    }
+
+    @Override
+    public UserRole getMinimalUserRole() {
+        return UserRole.ANONYMOUS;
```
Comment (dynamitechetan) [inline src/ai/susi/server/api/aaa/AddNewDevice.java]: This access level must be set to USER.

File: src/ai/susi/server/api/aaa/AddNewDevice.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            String s = result.toString();
+            byte b[]=s.getBytes();//converting string into byte array
+            fs.write(b);
+          }
+          catch(Exception e){
+            DAO.severe(e);
+          }
+
+          FileWriter fw = null;
+          BufferedWriter bw = null;
```
Comment (saurabhjn76) [inline src/ai/susi/server/api/aaa/AddNewDevice.java]: Move these declarations at the beginning of the method.

**Comment by Akshat-Jain** [comment]
@dynamitechetan @saurabhjn76 Made the required changes :-)

**Comment by Orbiter** [comment]
I see that here a new storage principle is introduced: a json file for each user ID. stored is the information email and an ID. I don't know why you want to store an email address using the user ID which is actually also an emails address (in most cases).

The whole API has no definition, which information shall be stored and in which data structure this shall happen. There is no abstraction layer of this information.

Furthermore, we already have a storage mechanism for user-related data: the accounting object. You should not store new information in a new data structure, instead create a class which handles your new data and adds it to the accounting object.

For an example on how to use the Accounting, see the class https://github.com/fossasia/susi_server/blob/development/src/ai/susi/server/api/aaa/ChangeUserSettings.java.

Please start from scratch: find out what information is needed to identify an device. Then make a data structure for this. Then investigate which API is already there and if you can use it.

**Pull request closed without merging**